presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -138,229 +75,39 @@ public OrcRecordReader(
-    }
-
-    /**
-     * Returns the total number of rows in the file. This count includes rows
-     * for stripes that were completely excluded due to stripe statistics.
-     */
-    public long getFileRowCount()
-    {
-        return fileRowCount;
-    }
-
-    /**
-     * Return the row position within the stripes being read by this reader.
-     * This position will include rows that were never read due to row groups
-     * that are excluded due to row group statistics. Thus, it will advance
-     * faster than the number of rows actually read.
-     */
-    public long getReaderPosition()
-    {
-        return currentPosition;
-    }
-
-    /**
-     * Returns the total number of rows that can possibly be read by this reader.
-     * This count may be fewer than the number of rows in the file if some
-     * stripes were excluded due to stripe statistics, but may be more than
-     * the number of rows read if some row groups are excluded due to statistics.
-     */
-    public long getReaderRowCount()
-    {
-        return totalRowCount;
-    }
-
-    public long getSplitLength()
-    {
-        return splitLength;
-    }
-
-    /**
-     * Returns the sum of the largest cells in size from each column
-     */
-    public long getMaxCombinedBytesPerRow()
-    {
-        return maxCombinedBytesPerRow;
-    }
-
-    @Override
-    public void close()
-            throws IOException
-    {
-        try (Closer closer = Closer.create()) {
-            closer.register(orcDataSource);
-            for (StreamReader column : streamReaders) {
-                if (column != null) {
-                    closer.register(() -> column.close());
-                }
-            }
-        }
-
-        if (writeChecksumBuilder.isPresent()) {
-            WriteChecksum actualChecksum = writeChecksumBuilder.get().build();
-            validateWrite(validation -> validation.getChecksum().getTotalRowCount() == actualChecksum.getTotalRowCount(), "Invalid row count");
-            List<Long> columnHashes = actualChecksum.getColumnHashes();
-            for (int i = 0; i < columnHashes.size(); i++) {
-                int columnIndex = i;
-                validateWrite(validation -> validation.getChecksum().getColumnHashes().get(columnIndex).equals(columnHashes.get(columnIndex)),
-                        "Invalid checksum for column %s", columnIndex);
-            }
-            validateWrite(validation -> validation.getChecksum().getStripeHash() == actualChecksum.getStripeHash(), "Invalid stripes checksum");
-        }
-        if (fileStatisticsValidation.isPresent()) {
-            List<ColumnStatistics> columnStatistics = fileStatisticsValidation.get().build();
-            writeValidation.get().validateFileStatistics(orcDataSource.getId(), columnStatistics);
-        }
-    }
-
-    public boolean isColumnPresent(int hiveColumnIndex)
-    {
-        return presentColumns.contains(hiveColumnIndex);
     }
 
     public int nextBatch()
```

Comment:
I think some of the function body can be extracted out as a common part for both readers. That can prevent many variables becoming "protected":
```java
int prepareNextBatch()
        throws IOException
{
    // update position for current row group (advancing resets them)
    filePosition += currentBatchSize;
    currentPosition += currentBatchSize;

    // if next row is within the current group return
    if (nextRowInGroup >= currentGroupRowCount) {
        // attempt to advance to next row group
        if (!advanceToNextRowGroup()) {
            filePosition = fileRowCount;
            currentPosition = totalRowCount;
            return -1;
        }
    }

    // We will grow currentBatchSize by BATCH_SIZE_GROWTH_FACTOR starting from initialBatchSize to maxBatchSize or
    // the number of rows left in this rowgroup, whichever is smaller. maxBatchSize is adjusted according to the
    // block size for every batch and never exceed MAX_BATCH_SIZE. But when the number of rows in the last batch in
    // the current rowgroup is smaller than min(prepareNextBatch, maxBatchSize), the prepareNextBatch for next batch in
    // the new rowgroup should be grown based on min(prepareNextBatch, maxBatchSize) but not by the number of rows in
    // the last batch, i.e. currentGroupRowCount - nextRowInGroup. For example, if the number of rows read for
    // single fixed width column are: 1, 16, 256, 1024, 1024,..., 1024, 256 and the 256 was because there is only
    // 256 rows left in this row group, then the prepareNextBatch should be 1024 instead of 512. So we need to grow the
    // prepareNextBatch before limiting the currentBatchSize by currentGroupRowCount - nextRowInGroup.
    currentBatchSize = toIntExact(min(nextBatchSize, maxBatchSize));
    nextBatchSize = min(currentBatchSize * BATCH_SIZE_GROWTH_FACTOR, MAX_BATCH_SIZE);
    currentBatchSize = toIntExact(min(currentBatchSize, currentGroupRowCount - nextRowInGroup));

    return currentBatchSize;
}

Reply:
Also, this `prepareNextBatch` can return a struct containing both `currentBatchSize` and `nextRowInGroup` (with `nextRowInGroup` updated) so that we can avoid `nextRowInGroup` becoming "protected"